model_server.py

```
@@ -0,0 +1,275 @@
+
+        return PredictionResponse(
+            predicted_class=predicted_class,
+            top_predictions=results,
+            electric_range_stats=range_stats,
+        )
+
+    except Exception as e:
+        import traceback
+
+        print(traceback.format_exc())  # Print full traceback for debugging
+        raise HTTPException(status_code=500, detail=f"Error processing image: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)